lbrynet/daemon/Daemon.py

```diff
@@ -1966,6 +1966,9 @@ def jsonrpc_publish(self, name, bid, metadata=None, file_path=None, fee=None, ti
         except (TypeError, URIParseError):
             raise Exception("Invalid name given to publish")
 
+        if not isinstance(bid, float):
+            raise Exception("Bid must be a float")
```

Comment:
Don't invent new exceptions unless it's somehow specific to the app. This particular exception looks like a TypeError, so just throw a TypeError. Or, pick something else more appropriate from here: https://docs.python.org/2/library/exceptions.html

Reply:
Sorry, this should be `ValueError`, not `TypeError`.

Reply:
Investigated this some more, `TypeError` is the correct exception:

`TypeError` - raised when a variable is of the wrong type. So, if you are expecting an int and it's a float, that's a **type** issue.

`ValueError` - is when you expect a string that you then want to convert to a float and it's an integer, then that's a value error. because string is correct **type**, but the **value** of the string is not valid.